### PR TITLE
Fix most of the warnings

### PR DESCRIPTION
**Pull request by knopers8**
The rest of the warnings are inside the imgui source files. Is there a way to ignore them?

### CODE REVIEW

File: Framework/include/QualityControl/ServiceDiscovery.h [hunk 1]
```diff
@@ -58,7 +59,11 @@ class ServiceDiscovery
   /// Health check thread loop
   void runHealthServer(unsigned int port);
 
-  static inline std::string GetDefaultUrl(); ///< Provides default health check URL
+  static inline std::string GetDefaultUrl() ///< Provides default health check URL
```
Comment (knopers8) [inline Framework/include/QualityControl/ServiceDiscovery.h]: To be inlined, a function must actually have its definition by the declaration.

File: Framework/src/DataDumpGui.cxx [hunk 2]
```diff
@@ -139,7 +139,6 @@ void updatePayloadGui()
     // header row
     ImGui::Separator();
     resizeColumns(representation /*, old_representation*/);
-    ImGui::Text("");
```
Comment (knopers8) [inline Framework/src/DataDumpGui.cxx]: I confirmed that it doesn't break the GUI

File: Framework/src/InformationService.h [hunk 3]
```diff
@@ -38,7 +38,7 @@ namespace pt = boost::property_tree;
 /// See runInformationService.cxx for the steering code.
 ///
 /// Example usage :
-///      qcInfoService -c /absolute/path/to/InformationService.json -n information_service \\
+///      qcInfoService -c /absolute/path/to/InformationService.json -n information_service
```
Comment (knopers8) [inline Framework/src/InformationService.h]: I couldn't think of another way to get rid of that warning, any ideas?

Comment (Barthelemy) [reply Framework/src/InformationService.h]: ANy ways, this is going to disappear as it is replaced by the ServiceDiscovery of Adam and George.

**Comment by knopers8** [comment]
The 'missing initializer for members' warnings will fixed in https://github.com/AliceO2Group/AliceO2/pull/2016 , if it is accepted

**Comment by Barthelemy** [comment]
Thank you very much. 
I get the following as well : 
```
In file included from /Users/bvonhall/dev/alice/sw/SOURCES/QualityControl/knopers8-fix-warnings/0/Framework/src/runBasic.cxx:66:
/Users/bvonhall/dev/alice/sw/SOURCES/QualityControl/knopers8-fix-warnings/0/Framework/src/ExamplePrinterSpec.h:34:41: warning: moving a temporary object prevents copy elision [-Wpessimizing-move]
    std::shared_ptr<TObjArray> moArray{ std::move(DataRefUtils::as<TObjArray>(*processingContext.inputs().begin())) };
                                        ^
```
Is it only on mac ?

**Comment by knopers8** [comment]
I haven't seen this one on CC7, I will fix it as well.

**Comment by knopers8** [comment]
The o2-dataflow build is stuck...

**Pull request merged**